### ISSUE TITLE
Add run Export

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/xtensor-feedstock#166]
### Comment:

Should this package (and xtl) have a run_export? it seems like yes.

**Comment by JohanMabille** [comment]
I am not sure of this. xtensor and xtl are header-only libraries, nothing should depend on them at runtime. So theoretically, their feedstock should not have a run_export section.

However, the following situation may be problematic:
- a library A depends on xtensor, and exposes xtensor objects in its API
- a library B depends on A and on xtensor.

When building B, we should have a mean to constrain the version of xtensor to be compatible with that used to build A. I think that should be a host dependency constraint, not a run constraint, but I 'm not sure we have this feature available in conda-forge. Having a run_export section would "workaound" this issue, but at the cost of installing xtensor whenever you install a library that depends on it at build time.

**Comment by hmaarrfk** [comment]
I see. Understood. I’ll leave this be for now and you can reassess later

**Comment by hmaarrfk** [comment]
> which issue / usecase you encounter that led you to open this discussion (sorry for that); would the run_constraint solve it?

I was probably helping a recipe that depended on xtensor and found:

```
  host:
    - xtl >=0.7.5,<0.8
  run:
    - xtl >=0.7.5,<0.8

**Comment by hmaarrfk** [comment]
Actually, this was it, it was the "nifty" https://github.com/conda-forge/nifty-feedstock/blob/15d5f0440ebe5af323009c21194801e74faefd1e/recipe/meta.yaml#L32

but if it is a header only library, then it is likely ok.

**Comment by JohanMabille** [comment]
Ok, so we should add a run_constraint on xtl in the xtensor recipe.

**Comment by hmaarrfk** [comment]
i don't think we have "really good" established practices for header only dependencies. so i'll let you decide and experiment.

I just found that kind of constraint "weird".
